atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -431,7 +431,7 @@ public long setCommitIndex(final long commitIndex) {
     if (commitIndex > previousCommitIndex) {
       this.commitIndex = commitIndex;
       raftLog.setCommitIndex(Math.min(commitIndex, raftLog.getLastIndex()));
-      if (raftLog.shouldFlushExplicitly() && isLeader()) {
+      if (isLeader()) {
```

Comment:
I think this is also a good example that it should be called `lastFlushedIndex` :smile: Then it makes sense to move it to the flusher, although I was hoping to keep the flusher implementations stateless. Let's see.

Reply:
I'm also starting to think this is purely journal metadata, not even Raft metadata. But we don't have a journal meta store or something :smile:

Reply:
When we added it, it was tied to raft because only committed indices are relevant. If there are indices that are flushed but not committed, then corruption on those records can be ignored. That was the idea. But with the changes in this PR, it is simpler to just update whenever it is flushed, instead restricting it to the commit index.

Reply:
Pushing that into the journal would simplify things here, since in the next PR with async flush, we will have to check for two potentially closed resources (e.g. Journal and MetaStore). Having one single resource (the journal) handle this would help. Otherwise we have to rely on carefully closing resources in the right order (Flusher before either Journal or MetaStore).

Reply:
This seems to be then a bigger change. Would also make sense to do it as a separate PR. So for now, a temporary fix by adding `raftLog.flushDirectly` here would be enough.